Assignment two

dspanesar opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

For this change, 2 files were modified the Assignment2.md and Assignment2.sql to complete assignment 2. The Assignment2.md was modified to complete Sections 1 and 4, the written response for the customer_address table options and the Ethics written response. While Assignment2.sql was modified to complete both Section 2 and Section 3, the coding components of the SQL assignment. Lastly a png file with a logical model was added to complete Section 1 of Assignment 2.

## What did you learn from the changes you have made?
Quite a lot was learned from this assignment. I had a chance to get comfortable with coalesce commands, windowed functions, string manipulations, regex, and unions for section 2. For section three I practiced my table modifications with cross joins, insert, delete and update. Lastly I learned about the implications of the human components of data and how that affects our ML and other models.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
n/a

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
I struggled a bit with both the delete operations i.e., how to frame them and the update operation. 
I googled a couple items and walked through the issues with our great technical facilitators during the Friday work period. 

## How were these changes tested?
I on several instances tested my code in the db sql lite program, made edits then tested again always with the provided db file.

## A reference to a related issue in your repository (if applicable)
n/a

## Checklist
- [x] I can confirm that my changes are working as intended

dspanesar committed:
Update Assignment2.md

Responses to posed assignment 2 questions (Sections 1 and 4)

dspanesar committed:
Add files via upload

dspanesar committed:
Update assignment2.sql

Assignment 2 submissions for Sections 2 and 3 SQL coding sections

kelichiu commented on 02_activities/assignments/assignment2.sql:
```diff
@@ -32,17 +37,64 @@ each new market date for each customer, or select only the unique market dates p
 HINT: One of these approaches uses ROW_NUMBER() and one uses DENSE_RANK(). */
 
+-- THIS SHOWS IF CUSTOMER VISITS MULTIPLE TIMES IN A DAY THE SAME RANK IS ASSIGNED TO EVERY ONE OF THE SAME MARKET DATES THE CUSTOMER SHOPPED ON
+SELECT *
```
You don't actually need this outer query.

dspanesar replied:
Thankyou for the feedback, I just wanted to make sure which query the FROM ( )x part?. 
Should I fix this and resubmit or is this okay as is?

kelichiu replied:
Hi Darshan, if you are happy with the grade then you don't need to resubmit

dspanesar replied:
I made the recommended corrections and resubmitting thankyou so much for the feedback. I merged the 2 pull requestions I believe.

kelichiu reviewed (commented):
## Total Point: 66/70

## Section 1: 44/46
- Descriptions for type 1 and type 2 are correct, but the other way.
- Date and Employee_Shifts’s relationship should be one to many


## Section 2: 8/8

## Section 3: 6/8
- DELETE:
	- (SELECT product_id FROM old_record: missing closing parentheses
## Section 4: 8/8

dspanesar committed:
Update Assignment2.md

I fixed the incorrectly labeled Type 1 and Type 2 tables for customer_address

dspanesar committed:
Rename LOGIC DIAGRAM ASSIGNMENT 2 BOOKSTORE - DSP.png to OLD-LOGIC DIAGRAM ASSIGNMENT 2 BOOKSTORE - DSP.png

Named OLD as corrected new diagram has been added per feedback

dspanesar committed:
Add files via upload

Corrected and updated diagram for Assignment 2 - per received feedback.

dspanesar committed:
Update assignment2.sql

Issues corrected per received feedback:

Removed unnecessary outer queries (Select, FROM and Brackets) in section 2. 

Also corrected the Delete section code which was missing a closing parentheses at the end.

kelichiu commented on 02_activities/assignments/assignment2.sql:
```diff
@@ -273,7 +269,7 @@ WHERE (product_id =
 AND snapshot_timestamp = 
 (SELECT snapshot_timestamp FROM old_record
```
You are missing the ) here, not the end of the query.

kelichiu reviewed (commented):
## Second Assessment: 70/70

## Section 1: 46/46


## Section 2: 8/8


## Section 3: 8/8


## Section 4: 8/8
